var/spack/repos/builtin/packages/py-pil/package.py

```diff
@@ -37,7 +37,13 @@ class PyPil(Package):
 
     provides('pil')
 
-    extends('python')
+    # py-pil currently only works with Python2.
+    # If you are using Python3, add to your packages.yaml:
+    # packages:
+    #     all:
+    #         providers:
+    #             pil: [py-pillow]
+    extends('python@:2.8')
```

Comment:
If you want to be even more specific, it supports Python 1.5.2 to 2.7. Not that I'm worried that someone is going to try to build it with Python 1.4 or older lol.